poetry_docker_plugin/docker_builder.py

```diff
@@ -283,80 +283,21 @@ def build(
-            # when there are no platforms specified, use standard build command
-            result = subprocess.run(
-                [
-                    "docker",
-                    "build",
-                    "--no-cache",
-                    *[f"--build-arg={arg}={value}" for arg, value in ({} if arguments is None else arguments).items()],
-                    *[arg for tag in image_tags for arg in ["--tag", tag]],
-                    "--file",
-                    f"dist/{dockerfile_name}",
-                    os.path.abspath("dist"),
-                ],
-                stdin=subprocess.PIPE,
-                stdout=subprocess.PIPE,
-                universal_newlines=True,
-            )
-        elif len(platform) < 2:
-            # when there is one platform specified, use cross-build command and load the image into docker
-            result = subprocess.run(
-                [
-                    "docker",
-                    "buildx",
-                    "build",
-                    "--load",
-                    "--no-cache",
-                    f"--platform={platform[0]}",
-                    *[f"--build-arg={arg}={value}" for arg, value in ({} if arguments is None else arguments).items()],
-                    *[arg for tag in image_tags for arg in ["--tag", tag]],
-                    "--file",
-                    f"dist/{dockerfile_name}",
-                    os.path.abspath("dist"),
-                ],
-                stdin=subprocess.PIPE,
-                stdout=subprocess.PIPE,
-                universal_newlines=True,
-            )
-        else:
-            # when there is more than one platform specified, use cross-build command and keep images in build cache
-            result = subprocess.run(
-                [
-                    "docker",
-                    "buildx",
-                    "build",
-                    "--no-cache",
-                    f"--platform={','.join(platform)}",
-                    *[f"--build-arg={arg}={value}" for arg, value in ({} if arguments is None else arguments).items()],
-                    *[arg for tag in image_tags for arg in ["--tag", tag]],
-                    "--file",
-                    f"dist/{dockerfile_name}",
-                    os.path.abspath("dist"),
-                ],
-                stdin=subprocess.PIPE,
-                stdout=subprocess.PIPE,
-                universal_newlines=True,
-            )
+        build_command = BuildCommand(image_tags, platform, arguments, dockerfile_name)
+        result = subprocess.run(
+            build_command.command(),
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            universal_newlines=True,
+        )
 
         if result.returncode == 0:
             self._io.write_line("<info>[INFO]:</info> Image tags successfully created!")
 
         if push and len(platform) > 1:
+            push_command = PushCommand(image_tags, platform, arguments, dockerfile_name)
```

Comment:
same thing.  created a class to create the push command to make it easy to test the command was being created.